components/brave_wallet_ui/components/extension/lock-panel/index.tsx

```diff
@@ -1,56 +1,99 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
 import * as React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+// utils
+import { getLocale } from '../../../../common/locale'
+
+// actions
+import { WalletActions } from '../../../common/actions'
+
+// types
+import { WalletState } from '../../../constants/types'
+
+// components
+import { PasswordInput } from '../../shared'
+import { NavButton } from '../'
 
+// style
 import {
   StyledWrapper,
   Title,
   Column,
   PanelIcon,
   RestoreButton
 } from './style'
-import { PasswordInput } from '../../shared'
-import { NavButton } from '../'
-import { getLocale } from '../../../../common/locale'
 
 export interface Props {
-  value?: string
-  onSubmit: () => void
-  onPasswordChanged: (value: string) => void
-  onClickRestore: () => void
-  hasPasswordError: boolean
-  disabled: boolean
+  onSubmit: (password: string) => void
+  onClickRestore?: () => void
+  disabled?: boolean
+  hideBackground?: boolean
 }
 
-function LockPanel (props: Props) {
-  const { value, onSubmit, onPasswordChanged, onClickRestore, disabled, hasPasswordError } = props
+export function LockPanel ({
+  onSubmit,
+  onClickRestore,
+  disabled,
+  hideBackground
+}: Props) {
+  // redux
+  const dispatch = useDispatch()
+  const hasIncorrectPassword = useSelector(({
+    wallet
+  }: { wallet: WalletState }) => wallet.hasIncorrectPassword)
+
+  // state
+  const [inputValue, setInputValue] = React.useState<string>('')
 
-  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+  // methods
+  const handlePasswordChanged = React.useCallback((value: string) => {
+    setInputValue(value)
+    if (hasIncorrectPassword) {
+      dispatch(WalletActions.hasIncorrectPassword(false))
+    }
+  }, [hasIncorrectPassword])
+
+  const handleKeyDown = React.useCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
     if (event.key === 'Enter' && !disabled) {
-      onSubmit()
+      onSubmit(inputValue)
     }
-  }
+  }, [disabled, onSubmit, inputValue])
+
+  const onClickUnlockButton = React.useCallback(() => {
+    onSubmit(inputValue)
+    setInputValue('')
+  }, [onSubmit, inputValue])
 
+  // render
   return (
-    <StyledWrapper>
+    <StyledWrapper hideBackground={hideBackground}>
```

Comment:
Lock panel can now be rendered within other components.
This component now internally tracks it's own state (password input changes)